filters/tee/teeloopback.go

```diff
@@ -0,0 +1,65 @@
+package tee
+
+import (
+	"context"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	teePredicate "github.com/zalando/skipper/predicates/tee"
+	"net/url"
+)
+
+const FilterName = "teeLoopback"
+
+type teeLoopbackSpec struct{}
+type teeLoopbackFilter struct {
+	teeKey string
+}
+
+func (t *teeLoopbackSpec) Name() string {
+	return FilterName
+}
+
+func (t *teeLoopbackSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) < 1 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+	teeKey, ok := args[0].(string)
+	if !ok || teeKey == "" {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+	return &teeLoopbackFilter{
+		teeKey,
+	}, nil
+}
+
+func NewTeeLoopback() filters.Spec {
+	return &teeLoopbackSpec{}
+}
+
+func (f *teeLoopbackFilter) Request(ctx filters.FilterContext) {
+	origRequest := ctx.Request()
+	// prevent the loopback to be executed indefinitely
+	teeRegistry, registryExists := ctx.Request().Context().Value(teePredicate.ContextTeeKey).(map[string]bool)
+	if !registryExists {
+		teeRegistry = map[string]bool{}
```

Comment:
To be more efficient for memory usage please use this as a set, which saves a bool from being stored (empty struct is a zero byte `struct{}` to get an instance of empty struct you use `struct{}{}`:
```
set := make(map[string]struct{})
set["foo"] = struct{}{}
if _, ok := set["foo"]; ok { .. } // has the key "foo"
if _, ok := set["bar"]; !ok { .. } // has no key "bar"
```